src/main/java/unicash/model/commons/Amount.java

```diff
@@ -29,6 +71,58 @@ public static boolean isValidAmount(double amount) {
         return amount >= 0.00;
     }
 
+    /**
+     * Returns true if a given amount is a non-negative value.
+     */
+    public static boolean isPositiveAmount(double amount) {
+        return amount >= 0.00;
+    }
+
+    /**
+     * Returns true if a given amount is a non-negative value.
+     */
+    public static boolean isWithinRange(double amount) {
+        return (amount > Integer.MIN_VALUE) && (amount < Integer.MAX_VALUE);
+    }
+
+    /**
+     * Returns true if a given amount has less than or equal to two decimal places.
+     */
+    public static boolean hasNoMoreThanTwoDecimalPlaces(double amount) {
+        String stringValue = Double.toString(amount);
+        int decimalIndex = stringValue.indexOf(".");
+
+        // If there's no decimal point, then it has no decimal places
+        if (decimalIndex == -1) {
+            return false;
+        }
+        int decimalCount = stringValue.length() - decimalIndex - 1;
+
+        return decimalCount > 2;
```

Comment:
Maybe instead of trying to do string manipulations this way, we can write a quick Regex to handle the checking instead? Much like how `Location` and `Category` does their checking against a String input? This way, it's easier to understand the behavior at a glance and we could expand the support decimal places without much issue in the future